sympy/functions/special/delta_functions.py

```diff
@@ -218,12 +234,25 @@ def eval(cls, arg):
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero and H0 != S.NaN:
+            return H0
 
-    def _eval_rewrite_as_Piecewise(self, arg):
+    def _eval_rewrite_as_Piecewise(self, *args):
+        if len(args) == 2:
+            arg, H0 = args[0], args[1]
+        else:
+            arg, H0 = args[0], None
         if arg.is_real:
-            return Piecewise((1, arg > 0), (S(1)/2, Eq(arg, 0)), (0, True))
+            if H0 == None:
+                return Piecewise((1, arg > 0), (0, arg < 0))
```

Comment:
Actually sorry for that. Just replace error message with a normal message in my previous comment.  
Actually, what I mean is if one call `Heaviside(0)` then just output `S.NaN` and provide a message like "user should define a value for `Heaviside(0)` ".